### PR TITLE
CS: apply phpdoc_annotation_without_dot reworked

### PR DESCRIPTION
**Pull request by keradus**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | n/a |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

`phpdoc_annotation_without_dot` rule was requested by Symfony, most cases was adjusted in code, but the rule was a bit problematic to apply it everywhere.
This PR is based on extended implementation of that rule (see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2247). I think it covers the problematic cases properly, but let us discuss.
Also, for that reason, I decided to put it in independent PR.

### CODE REVIEW

**Comment by wouterj** [comment]
Is there a reason to capitialize some descriptions, but to lowercase others? It's seems quite inconsistent to me.

**Comment by keradus** [comment]
The intention for original rule was that the annotation description should not be a sentence, so we trim trailing dot. And if it's not a sentence, the leading char shouldn't be capitalized.
The fixer is not touching multi-sentence description, as it would break grammar.

**Comment by wouterj** [comment]
Hmm, alright. Still it's weird that one-line descriptions like https://github.com/symfony/symfony/pull/20366/files#diff-70d56b3a86265721eccd933af82b7332R113 aren't fixed. Maybe the fixer only checks for final stops and not capitialized words?

File: src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterMappingsPass.php [hunk 1]
```diff
@@ -108,15 +108,15 @@
      *
      * @param Definition|Reference $driver                  Driver DI definition or reference
      * @param string[]             $namespaces              List of namespaces handled by $driver
-     * @param string[]             $managerParameters       List of container parameters that could
-     *                                                      hold the manager name.
+     * @param string[]             $managerParameters       list of container parameters that could
```
Comment (xabbuh) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterMappingsPass.php]: I am okay with removing the dot, but changing the case here makes the docblocks very inconsistent as most places start with a capital letter.

Comment (xabbuh) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterMappingsPass.php]: I fear that we will then have many changes that will cause lots of conflicts when merging branches.

Comment (fabpot) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/CompilerPass/RegisterMappingsPass.php]: Indeed, we won't merge PRs that makes merging difficult.

**Comment by nicolas-grekas** [comment]
:-1: for me, unproven positive value, clear negative one (merge conflicts)

**Comment by keradus** [comment]
ref https://github.com/symfony/symfony/pull/19198

**Comment by nicolas-grekas** [comment]
Closing as won't fix: as commented, that'll creat merge conflicts - a final dot on comments doesn't deserve it (especially since we have so many PRs pending ;) )

**Comment by keradus** [comment]
@nicolas-grekas 
thanks for update

Could you please specify is it a rule that would be nice to apply in new code, but not old (for the conflicts reason), or no willingness to follow it in new code as well ? Especially, as the rule was initially requested by Symfony.

**Comment by nicolas-grekas** [comment]
@keradus not sure I can tell you really... Right now, we have E_TOO_MANY_PR, so I've been a bit strict here, to help clean the backlog. The fix might happen later I guess (but would be great to have fabbot not patch the case of the first letter first :) )

**Comment by keradus** [comment]
So, some CS rules that Symfony wants, but not apply them, as of conflicts with merging. Cool to loosely follow it on new PR, but not in old, existing code.
Would be nice to extract that kind of rules to separated group somehow instead of dropping of forgot about them ;) If not, there is no much reasoning to prepare that kind of requested rules, if they would not be used after all.

**Pull request closed without merging**